### PR TITLE
docs(contributing): include steps to preview changes in an external app 

### PR DESCRIPTION
**Pull request by brandyscarney**
updates the contributing guide to:

- include steps to package changes and test them in an external app
- fix the references to the test apps so they point to the right directories
- link to the test app directories for steps to run them and preview changes from within this repository

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -123,12 +123,126 @@ Before creating a pull request, please read our requirements that explains the m
...
+
+###### JavaScript
+
+Run the following commands to build the core directory and pack the changes:
+
+```bash
+cd core
+npm i
+npm run build
+npm pack --pack-destination ~
```
Comment (liamdebeasi) [inline .github/CONTRIBUTING.md:146]: Will these instructions work on Windows too? Alternatively, we could remove `--pack-destination ~` and use `[PATH TO TGZ]` or something in the `npm install` step below if the steps don't work on Windows.

Comment (brandyscarney) [reply .github/CONTRIBUTING.md:146]: Good point! I will need to fire up the Windows machine to test this

Comment (liamdebeasi) [reply .github/CONTRIBUTING.md:146]: Sorry for making you fire up the Windows machine 😂

Comment (amandaejohnston) [reply .github/CONTRIBUTING.md:146]: If it helps, there's a bash terminal that I think comes packaged with Git, and VS Code has it built-in. I'm on a Windows machine myself and I do occasionally have to run scripts in bash for them to work, so there's precedent for that.

**Review by liamdebeasi — APPROVED** [review]

**Pull request merged**